sources/packages/backend/apps/api/src/route-controllers/overaward/overaward.controller.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from "@nestjs/common";
+import { DisbursementOverawardService } from "@sims/services";
+import { DisbursementOveraward } from "@sims/sims-db";
+import { getUserFullName } from "../../utilities";
+import {
+  OverawardAPIOutDTO,
+  OverawardBalanceAPIOutDTO,
+} from "./models/overaward.dto";
+
+/**
+ * Overaward controller service.
+ */
+@Injectable()
+export class OverawardControllerService {
+  constructor(
+    private readonly disbursementOverawardService: DisbursementOverawardService,
+  ) {}
+
+  /**
+   * Get the overaward balance of a student.
+   * @param studentId student.
+   * @returns overaward balance for student.
+   */
+  async getOverawardBalance(
+    studentId?: number,
+  ): Promise<OverawardBalanceAPIOutDTO> {
+    const overawardBalance =
+      await this.disbursementOverawardService.getOverawardBalance([studentId]);
+
+    return { overawardBalanceValues: overawardBalance[studentId] };
+  }
+
+  /**
+   * Get all overawards which belong to a student.
+   * @param studentId student.
+   * @param includeAddedBy include added by.
+   * @returns overaward details of a student.
+   */
+  async getOverawardsByStudent(
+    studentId?: number,
+    includeAddedBy = false,
+  ): Promise<OverawardAPIOutDTO[]> {
+    const overawards =
+      await this.disbursementOverawardService.getOverawardsByStudent(studentId);
+    return overawards.map((overaward) =>
+      this.transformOverawards(overaward, includeAddedBy),
+    );
+  }
+
+  /**
+   * Transform overawards for a student.
+   * @param overaward overaward.
+   * @param includeAddedBy include added by.
+   * @returns transformed overawards.
+   */
+  private transformOverawards(
+    overaward: DisbursementOveraward,
+    includeAddedBy: boolean,
+  ): OverawardAPIOutDTO {
+    return {
+      dateAdded: overaward.addedDate,
```

Comment:
I guess that there is a gap here from some previous PR. The date should come from the `overaward.addedDate` if it was a manually added record, but if it was created by the system I would assume that it should be the `created_at`, otherwise the column will be empty on the UI (and it also makes sense from API perspective).
Alternatively, the API can return both dates and the UI can decide which one to show.
@ann-aot @andrepestana-aot @dheepak-aot @sh16011993 thoughts?

Reply:
IMO, "the API can return both dates and the UI can decide which one to show" is simpler.

Reply:
Agree with @andrepestana-aot

Reply:
Yes done with the changes. Thanks for the confirmation team.